### ISSUE TITLE
`show` fails for a nonparametric model on short chains

### ISSUE THREAD

**Issue by phipsgabler** [TuringLang/MCMCChains.jl#237]
For the following model:

```julia
function update_histogram!(histogram, bin)
    if bin > length(histogram)
        push!(histogram, 1)
    else
        histogram[bin] += 1
    end

    return histogram
end

@model function imm(x)
    N = length(x)

    nk = Int[]
    K = 0
    z = zeros(Int, N)

    for n = 1:N
        z[n] ~ ChineseRestaurantProcess(DirichletProcess(1.0), nk)
        nk = update_histogram!(nk, z[n])
        K = max(K, z[n])
    end

    μ ~ filldist(Normal(), K)

    for n = 1:N
        x[n] ~ Normal(μ[z[n]], 1.0)
    end
end
```

I often get the followin error with very few samples:

```julia
julia> s = sample(model_imm, Prior(), 5)
Sampling 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| Time: 0:00:00
Object of type Chains, with data of type 5×7×1 Array{Union{Missing, Float64},3}

Iterations        = 1:5
Thinning interval = 1
Chains            = 1
Samples per chain = 5
internals         = lp
parameters        = z[1], z[2], z[3], μ[1], μ[2], μ[3]

Error showing value of type Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}:
ERROR: BoundsError: attempt to access 0-element Array{Any,1} at index [1]
Stacktrace:
 [1] getindex(::Array{Any,1}, ::Int64) at ./array.jl:744
 [2] #ess#200(::Bool, ::Array{Symbol,1}, ::Int64, ::Int64, ::Bool, ::typeof(ess), ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at /home/philipp/.julia/packages/MCMCChains/pVyO1/src/stats.jl:376
 [3] #ess at ./none:0 [inlined]
 [4] #summarystats#215(::Bool, ::Bool, ::Array{Symbol,1}, ::Symbol, ::Int64, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(summarystats), ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at /home/philipp/.julia/packages/MCMCChains/pVyO1/src/stats.jl:441
 [5] (::StatsBase.var"#kw##summarystats")(::NamedTuple{(:showall, :sections, :etype, :digits, :sorted),Tuple{Bool,Array{Symbol,1},Symbol,Int64,Bool}}, ::typeof(summarystats), ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at ./none:0
 [6] #describe#192(::Array{Float64,1}, ::Symbol, ::Bool, ::Array{Symbol,1}, ::Int64, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(describe), ::Base.TTY, ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at /home/philipp/.julia/packages/MCMCChains/pVyO1/src/stats.jl:203
 [7] describe(::Base.TTY, ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at /home/philipp/.julia/packages/MCMCChains/pVyO1/src/stats.jl:203
 [8] #describe#191(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(describe), ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at /home/philipp/.julia/packages/MCMCChains/pVyO1/src/stats.jl:174
 [9] describe at /home/philipp/.julia/packages/MCMCChains/pVyO1/src/stats.jl:174 [inlined]
 [10] show(::IOContext{REPL.Terminals.TTYTerminal}, ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at /home/philipp/.julia/packages/MCMCChains/pVyO1/src/chains.jl:283
 [11] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::Chains{Union{Missing, Float64},Missing,NamedTuple{(:internals, :parameters),Tuple{Array{String,1},Array{String,1}}},NamedTuple{(),Tuple{}}}) at ./multimedia.jl:47
 [12] display(::REPL.REPLDisplay, ::MIME{Symbol("text/plain")}, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/REPL/src/REPL.jl:132
 [13] display(::REPL.REPLDisplay, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.3/REPL/src/REPL.jl:136
 [14] display(::Any) at ./multimedia.jl:323
```

The same happens with `MH` as a sampler (although `MH` errors it in other ways as well).  When the number of samples is large, it almost never happens, though.  Or am I just using the CRP wrongly?  @trappmartin ?

**Comment by devmotion** [comment]
I'm a bit confused by your stacktrace - it seems to point to line 376 in `src/stats.jl` but actually in the most recent versions this file does not contain 376 lines anymore. In some older versions the ESS computation errored if not enough samples were available but this should be fixed in the more recent versions.

**Comment by phipsgabler** [comment]
Oh, right, I have pinned Turing and DPPL.  This is using MCMCChains 3.0.12.  If anyone can confirm it's fixed, lets close this.

**Comment by cpfiffer** [comment]
Yeah, this should be fixed with #229.
